pkg/config/ext_authz_test.go

```diff
@@ -0,0 +1,316 @@
+						ProtocolConfig: &httpOptions.HttpProtocolOptions_ExplicitHttpConfig_HttpProtocolOptions{},
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := extAuthzCluster(tt.args.host, tt.args.port, tt.args.protocol)
+			httpProtocolOptionsGot, ok := got.TypedExtensionProtocolOptions[extAuthzClusterTypedExtensionProtocolOptionsHTTP]
+
+			if !ok {
+				t.Errorf("Cannot found %s in config", extAuthzClusterTypedExtensionProtocolOptionsHTTP)
+			}
+
+			httpProtocolOptionsWantedAny, err := anypb.New(tt.httpProtocolOptionsWanted)
+			if err != nil {
+				t.Errorf("Cannot convert HttpProtocolOptions to Any")
+			}
+
+			if !reflect.DeepEqual(httpProtocolOptionsGot, httpProtocolOptionsWantedAny) {
+				t.Errorf("extAuthzCluster() http protocol options = %v, wanted %v", httpProtocolOptionsGot, httpProtocolOptionsWantedAny)
+			}
+		})
+	}
+}
+
+func Test_externalAuthZFilter_extAuthz(t *testing.T) {
+	tests := []struct {
+		name           string
+		conf           *config
+		extAuthzWanted *extAuthService.ExtAuthz
+	}{
+		{
+			name: "grpc",
+			conf: &config{
+				Host:            "example.com:50051",
+				MaxRequestBytes: 8192,
+				Timeout:         2000,
+				Protocol:        "grpc",
+			},
+			extAuthzWanted: &extAuthService.ExtAuthz{
+				TransportApiVersion: core.ApiVersion_V3,
+				WithRequestBody: &extAuthService.BufferSettings{
+					MaxRequestBytes:     8192,
+					AllowPartialMessage: true,
+				},
+				Services: &extAuthService.ExtAuthz_GrpcService{
+					GrpcService: &core.GrpcService{
+						TargetSpecifier: &core.GrpcService_EnvoyGrpc_{
+							EnvoyGrpc: &core.GrpcService_EnvoyGrpc{
+								ClusterName: extAuthzClusterName,
+							},
+						},
+						Timeout: durationpb.New(time.Duration(2000) * time.Millisecond),
+						InitialMetadata: []*core.HeaderValue{{
+							Key:   "client",
+							Value: "kourier",
+						}},
+					},
+				},
+			},
+		},
+		{
+			name: "http",
+			conf: &config{
+				Host:            "example.com:8080",
+				MaxRequestBytes: 8192,
+				Timeout:         2000,
+				Protocol:        "http",
+			},
+			extAuthzWanted: &extAuthService.ExtAuthz{
+				TransportApiVersion: core.ApiVersion_V3,
+				WithRequestBody: &extAuthService.BufferSettings{
+					MaxRequestBytes:     8192,
+					AllowPartialMessage: true,
+				},
+				Services: &extAuthService.ExtAuthz_HttpService{
+					HttpService: &extAuthService.HttpService{
+						ServerUri: &core.HttpUri{
+							Uri: "http://example.com:8080",
+							HttpUpstreamType: &core.HttpUri_Cluster{
+								Cluster: extAuthzClusterName,
+							},
+							Timeout: durationpb.New(time.Duration(2000) * time.Millisecond),
+						},
+						AuthorizationRequest: &extAuthService.AuthorizationRequest{
+							HeadersToAdd: []*core.HeaderValue{{
+								Key:   "client",
+								Value: "kourier",
+							}},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "http_with_path_prefix",
```

Comment:
```suggestion
			name: "http with path prefix",
```
It is fine to use spalce.